### PR TITLE
slim client dependencies

### PR DESCRIPTION
**Pull request by Ongy**
This removes the wlroots dependency for clients.
The clients used wlr_log, to get rid of the dependency, the relevant
macros and functions are copied over and renamed to sway_log.
This also "fixes" that wlr_strip_path strips the wlr source path,
sway_strip_path will strip the sway source directroy.
It also makes the ipc an explicit dependency, not part of sway_common,
to allow building clients that don't require IPC without jsonc.

### CODE REVIEW

File: swayipc/meson.build [hunk 1]
```diff
@@ -0,0 +1,8 @@
+lib_sway_ipc = static_library(
```
Comment (emersion) [inline swayipc/meson.build]: I'd name the IPC library just "ipc" (instead of "swayipc") or `common/ipc`, since sway* directories contain executables.

**Comment by emersion** [comment]
This LGTM overall. This is a good first step towards #1875.

I wonder if we should continue to use `wlr_log` for sway itself.

**Comment by ascent12** [comment]
I always thought having `wlr_log` itself as part of our API was a bit weird.
Originally, the user (sway) would use `wlr_log_init` into their own logging function.

**Comment by emersion** [comment]
Yeah, that's true. +1 for doing this.

File: include/log.h [hunk 2]
```diff
@@ -1,16 +1,52 @@
...
 
 void error_handler(int sig);
 
+typedef enum {
+	L_SILENT = 0,
+	L_ERROR = 1,
+	L_INFO = 2,
+	L_DEBUG = 3,
+	L_LAST,
+} log_importance_t;
```
[the remark this answers is not in the document]
Comment (Ongy) [reply include/log.h]: redefining errors. Which I'm not happy about, but I'm not aware of a better way

[the remark this answers is not in the document]
Comment (emersion) [reply include/log.h]: I mean: we should probably prefix this in wlroots with `wlr_` right?

**Comment by SirCmpwn** [comment]
You need to split this up into several commits so it can be properly reviewed.

File: include/log.h (hunk 2 above)
[the remark this answers is not in the document]
Comment (Ongy) [reply include/log.h]: Uff, they are just defines, so unless both headers are included it won't colide.
On the other hand that would make it easier to init things properly

File: sway/main.c [hunk 3]
```diff
@@ -336,21 +337,24 @@ int main(int argc, char **argv) {
 	// TODO: switch logging over to wlroots?
```
Comment (SirCmpwn) [inline sway/main.c]: This comment can probably go away

**Comment by SirCmpwn** [comment]
Part of me is saddened by this patch. But part of me accepts it as necessary. If you don't mind killing that dead comment, I'll get this merged up.

File: sway/main.c [hunk 4]
```diff
@@ -223,34 +223,35 @@ static void executable_sanity_check() {
...
 			cap_set_proc(caps)) {
-			wlr_log(L_ERROR, "Failed to drop extra capabilities");
+			sway_log(L_ERROR, "Failed to drop extra capabilities");
 			exit(EXIT_FAILURE);
 		}
 	}
 #endif
 }
 
+void wlr_log_init(log_importance_t verbosity, log_callback_t callback);
```
Comment (emersion) [inline sway/main.c]: Should include the appropriate header instead of doing this

Comment (Ongy) [reply sway/main.c]: that conflicts on the `L_DEBUG` etc. defines =.=

Comment (emersion) [reply sway/main.c]: Yeah, that's why we need to prefix wlroots symbols.

**Comment by emersion** [comment]
You should be able to use https://github.com/swaywm/wlroots/pull/1011 to properly fix this PR - let me know if you want me to help with that.

**Comment by emersion** [comment]
Closing, there are too many conflicts and sway will continue to use `wlr_log`.

**Pull request closed without merging**